captum/optim/_param/image/images.py

```diff
@@ -1,538 +1,457 @@
+
+
+class InputParameterization(torch.nn.Module):
+    def forward(self):
+        raise NotImplementedError
+
+
+class ImageParameterization(InputParameterization):
+    def setup_batch(
+        self, x: torch.Tensor, batch: int = 1, dim: int = 3
+    ) -> torch.Tensor:
+        assert batch > 0
+        x = x.unsqueeze(0) if x.dim() == dim and batch == 1 else x
+        x = (
+            torch.stack([x.clone() for b in range(batch)])
+            if x.dim() == dim and batch > 1
+            else x
+        )
+        return x
+
+
+class FFTImage(ImageParameterization):
+    """Parameterize an image using inverse real 2D FFT"""
+
+    def __init__(
+        self,
+        size: Tuple[int, int] = None,
+        channels: int = 3,
+        batch: int = 1,
+        init: Optional[torch.Tensor] = None,
+    ) -> None:
+        super().__init__()
+        if init is None:
+            assert len(size) == 2
+            self.size = size
+        else:
+            assert init.dim() == 3 or init.dim() == 4
+            self.size = (
+                (init.size(1), init.size(2))
+                if init.dim() == 3
+                else (init.size(2), init.size(3))
+            )
+        self.torch_rfft, self.torch_irfft = self.get_fft_funcs()
+
+        frequencies = FFTImage.rfft2d_freqs(*self.size)
+        scale = 1.0 / torch.max(
+            frequencies,
+            torch.full_like(frequencies, 1.0 / (max(self.size[0], self.size[1]))),
+        )
+        scale = scale * ((self.size[0] * self.size[1]) ** (1 / 2))
+        spectrum_scale = scale[None, :, :, None]
+        self.register_buffer("spectrum_scale", spectrum_scale)
+
+        if init is None:
+            coeffs_shape = (channels, self.size[0], self.size[1] // 2 + 1, 2)
+            random_coeffs = torch.randn(
+                coeffs_shape
+            )  # names=["C", "H_f", "W_f", "complex"]
+            fourier_coeffs = random_coeffs / 50
+        else:
+            fourier_coeffs = self.torch_rfft(init) / spectrum_scale
+
+        fourier_coeffs = self.setup_batch(fourier_coeffs, batch, 4)
+        self.fourier_coeffs = nn.Parameter(fourier_coeffs)
+
+    @staticmethod
+    def rfft2d_freqs(height: int, width: int) -> torch.Tensor:
+        """Computes 2D spectrum frequencies."""
+        fy = FFTImage.pytorch_fftfreq(height)[:, None]
+        # on odd input dimensions we need to keep one additional frequency
+        wadd = 2 if width % 2 == 1 else 1
+        fx = FFTImage.pytorch_fftfreq(width)[: width // 2 + wadd]
+        return torch.sqrt((fx * fx) + (fy * fy))
+
+    @staticmethod
+    def pytorch_fftfreq(v: int, d: float = 1.0) -> torch.Tensor:
+        """PyTorch version of np.fft.fftfreq"""
+        results = torch.empty(v)
+        s = (v - 1) // 2 + 1
+        results[:s] = torch.arange(0, s)
+        results[s:] = torch.arange(-(v // 2), 0)
+        return results * (1.0 / (v * d))
+
+    def get_fft_funcs(self) -> Tuple[Callable, Callable]:
```

Comment:
Would it possible to improve this function by using `torch.__version__ <= "1.7.0"` instead of a try except block?

Reply:
Updated!